keps/sig-node/2400-node-swap/README.md

```diff
@@ -0,0 +1,632 @@
+
+## Motivation
+
+There are two distinct types of user for swap, who may overlap:
+- node administrators, who may want swap available for node-level performance tuning and stability/reducing noisy neighbour issues
+- application developers, who have written applications that would benefit from using swap memory
+
+There are hence a number of possible ways that one could envision swap use on a node.
+
+### Scenarios 
+
+1. Swap is enabled only at the system level. The CRI does not permit user workloads to use swap. (This scenario is a prerequisite for the following use cases.)
+1. Swap is enabled at the node level. The CRI can be globally configured to permit user workloads scheduled on the node to use some quantity of swap.
+1. Swap is set on a per-workload basis. The CRI sets permitted swap utilization on each individual workload.
+
+This KEP will be limited in scope to the first two scenarios. The third can be addressed in a follow-up KEP. The enablement work that is in scope for this KEP will be necessary to implement the third scenario.
+
+
+### Goals
+
+- On Linux systems, when swap is provisioned and available, Kubelet can start up with swap on.
+- Configuration is available for CRI to set swap utilization available to Kubernetes workloads, defaulting to 0 swap.
+- Cluster administrators can enable and configure CRI swap utilization on a per-node basis.
+
+### Non-Goals
+
+- Provisioning swap. Swap must already be available on the system.
+- Allocating swap on a per-workload basis with accounting (e.g. pod-level specification of swap). If desired, this should be designed and implemented as part of a follow-up KEP. This KEP is a prerequisite for that work.
+- Supporting zram, zswap, or other memory types like SGX EPC. These could be addressed in a follow-up KEP, and are out of scope.
+
+## Proposal
+
+I propose that, when swap is provisioned and available on a node, we allow cluster administrators to configure the Kubelet and CRI such that:
+
+- The kubelet can start with swap on.
+- The CRI is updated such that by default, workloads will use 0 swap.
+- The CRI will have configuration available such that swap utilization can be configured for the entire node (e.g. as a percentage of pod memory requests).
+
+This proposal enables scenarios 1 and 2 above, but not 3.
+
+### User Stories
+
+#### Improved Node Stability
+
+cgroupsv2 improved memory management algos, such as oomd, currently require swap. Hence, having a small amount of swap available on nodes could improve better resource pressure handling and recovery.
+
+- https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#id1 
+- https://chrisdown.name/2018/01/02/in-defence-of-swap.html 
+- https://media.ccc.de/v/ASG2018-175-oomd 
+- https://github.com/facebookincubator/oomd/blob/master/docs/production_setup.md#swap 
+
+This user story is addressed by scenario 1 and 2, and could benefit from 3.
+
+#### Long-running applications that swap out startup memory
```

Comment:
can we also mention the machine learning workloads which load large datasets for processing?